### PR TITLE
Index blocks and transactions for API node

### PR DESCRIPTION
**Pull request by ggicci**

### CODE REVIEW

File: blockproducer/branch.go [hunk 1]
```diff
@@ -172,20 +174,28 @@ func (b *branch) produceBlock(
...
+		ierr      error
+		packCount = transactionsLimit
 	)
-	for _, v := range txs {
+
+	if len(txs) < packCount {
+		packCount = len(txs)
+	}
+
+	out := make([]pi.Transaction, len(txs))
```
Comment (xq262144) [inline blockproducer/branch.go]: Should use `packCount` instead of `len(txs)`

File: blockproducer/branch.go [hunk 2]
```diff
@@ -172,20 +174,28 @@ func (b *branch) produceBlock(
...
+		packCount = transactionsLimit
 	)
-	for _, v := range txs {
+
+	if len(txs) < packCount {
+		packCount = len(txs)
+	}
+
+	out := make([]pi.Transaction, packCount)
+	for i := 0; i < packCount; i++ {
```
Comment (leventeliu) [inline blockproducer/branch.go]: Should also verify block tx number limit at `pushBlock`.
See `branch.applyBlock` and `branch.go: newBranch`.

**Review by leventeliu — APPROVED** [review]
LGTM.

**Review by xq262144 — APPROVED** [review]
Approve

**Pull request merged**